src/Rotations.jl

```diff
@@ -0,0 +1,318 @@
+
+This QR-based retraction is a first-order approximation of the exponential map.
+"""
+function retract!(M::Rotations, y::AbstractArray{T}, x, v, method::QRRetraction) where T
+    A = x + x*v
+    qr_decomp = qr(A)
+    d = diag(qr_decomp.R)
+    D = Diagonal( sign.(d .+ convert(T, 0.5)) )
+    y .= qr_decomp.Q * D
+    return y
+end
+
+retract!(M::Rotations, y, x, v) = retract!(M, y, x, v, QRRetraction())
+
+"""
+    PolarInverseRetraction
+
+Inverse retraction on the rotations manifold using the polar method.
+"""
+struct PolarInverseRetraction <: AbstractInverseRetractionMethod end
+
+injectivity_radius(::Rotations, x, ::PolarInverseRetraction) = π*sqrt(2.0)/2
+
+@doc doc"""
+    inverse_retract!(M, v, x, y, ::PolarInverseRetraction)
+
+Compute a vector from the tagent space $T_x\mathrm{SO}(n)$
+of the point `x` on the [`Rotations`](@ref) manifold `M`
+with which the point `y` can be reached by the
+[`PolarRetraction`](@ref) from the point `x` after time 1.
+
+The formula reads
+$v = -\frac{1}{2}(x^{\mathrm{T}}ys - (x^{\mathrm{T}}ys)^{\mathrm{T}})$
+
+where $s$ is the solution to the Sylvester equation
+
+$x^{\mathrm{T}}ys + s(x^{\mathrm{T}}y)^{\mathrm{T}} + 2\mathrm{I}_n = 0.$
+"""
+function inverse_retract!(M::Rotations, v, x, y, method::PolarInverseRetraction)
+    A = transpose(x) * y
+    H = 2 * one(x)
+    try
+        B = sylvester(collect(A), collect(transpose(A)), collect(H))
+        C = A * B
+        v .= (transpose(C) .- C)./2
+    catch e
+        if isa(e, LinearAlgebra.LAPACKException)
+            throw(OutOfInjectivityRadiusError())
+        else
+            rethrow()
+        end
+    end
+    return v
+end
+
+"""
+    QRInverseRetraction
+
+Inverse retraction on the rotations manifold using the QR method.
+"""
+struct QRInverseRetraction <: AbstractInverseRetractionMethod end
+
+@doc doc"""
+    inverse_retract!(M::Rotations, x, y, ::QRInverseRetraction)
+
+Compute a vector from the tagent space $T_x\mathrm{SO}(n)$
+of the point `x` on the [`Rotations`](@ref) manifold `M`
+with which the point `y` can be reached by the
+[`QRRetraction`](@ref) from the point `x` after time 1.
+"""
+function inverse_retract!(M::Rotations{N}, v, x, y, ::QRInverseRetraction) where N
+    A = transpose(x) * y
+    R = zero(v)
+    for i = 1:N
+        b = zeros(i)
+        b[end] = 1
+        b[1:(end-1)] = - transpose(R[1:(i-1), 1:(i-1)]) * A[i, 1:(i-1)]
+        R[1:i, i] = A[1:i, 1:i] \ b
+    end
+    C =  A * R
+    v .= (C .- transpose(C))./2
+    return v
+end
+
+zero_tangent_vector(S::Rotations, x) = zero(x)
+zero_tangent_vector!(S::Rotations, v, x) = (v .= zero(x))
+
+"""
+    is_manifold_point(S,x; kwargs...)
+
+checks, whether `x` is a valid point on the [`Rotations`](@ref) `S`,
+i.e. is an array of size [`manifold_dimension`](@ref)`(S)` and represents a
+valid rotation.
+The tolerance for the last test can be set using the ´kwargs...`.
+"""
+function is_manifold_point(S::Rotations{N},x; kwargs...) where {N}
```

Comment:
I would expect a function called `is_manifold_point` to return `true` if the point is on the manifold and `false` if not. Perhaps instead of throwing an error, this function should do that, and then a function named something like `check_manifold_point` would call this and throw an error if `false`.

Reply:
The function does the same thing for `Sphere`, I'll make a separate PR to change and discuss this.